app/services/IncomeSourceDetailsService.scala

```diff
@@ -120,6 +121,54 @@ class IncomeSourceDetailsService @Inject()(val incomeTaxViewChangeConnector: Inc
     }.toEither
   }
 
+  def getViewIncomeSourceViewModel(sources: IncomeSourceDetailsModel): Either[Throwable, ViewIncomeSourcesViewModel] = {
+
+    val maybeSoleTraderBusinesses = sources.businesses.filterNot(_.isCeased)
+    val soleTraderBusinessesExists = maybeSoleTraderBusinesses.nonEmpty
+
+    val maybeUkProperty = sources.properties.find(_.isUkProperty)
+    val ukPropertyExists = maybeUkProperty.nonEmpty
+
+    val maybeForeignProperty = sources.properties.find(_.isForeignProperty)
+    val foreignPropertyExists = maybeForeignProperty.nonEmpty
+
+    val maybeCeasedBusinesses = sources.businesses.filter(_.isCeased)
+    val ceasedBusinessExists = maybeCeasedBusinesses.nonEmpty
+
+    Try {
+      ViewIncomeSourcesViewModel(
+        viewSoleTraderBusinesses = if (soleTraderBusinessesExists) {
+          maybeSoleTraderBusinesses.map { business =>
+            ViewBusinessDetailsViewModel(
+              business.tradingName.getOrElse("Unknown"),
+              business.tradingStartDate
+            )
+          }
+        } else Nil,
+        viewUkProperty = if (ukPropertyExists) {
+          Some(ViewPropertyDetailsViewModel(
+            maybeUkProperty.flatMap(_.tradingStartDate)
+          ))
+        } else None,
+        viewForeignProperty = if (foreignPropertyExists) {
+          Some(ViewPropertyDetailsViewModel(
+            maybeForeignProperty.flatMap(_.tradingStartDate)
+          ))
+        } else None,
+        viewCeasedBusinesses = if (ceasedBusinessExists) {
+          maybeCeasedBusinesses.map { business =>
+            ViewCeasedBusinessDetailsViewModel(
+              tradingName = business.tradingName.getOrElse("Unknown"),
+              tradingStartDate = business.tradingStartDate,
+              cessationDate = business.cessation.flatMap(_.date).getOrElse(LocalDate.parse("Unknown"))
```

Comment:
Unsure how to display the unknown for business name when throwing the exception if its mandatory field

Reply:
we had a discussion yesterday regarding similar cases;
Here is what we have agreed (this would be similar for any field where we have to display 'Unknown'): 
* set tradingName as Optional in the view model, so no need to throw error when you map API returned field to view model field;
* in the actual view, do next call tradingName.getElse(  @getMessage("~.relevant.resource.for.nnknown") )

so, we would show something in case there is no value provided for this field etc